cmd/go-tpc/main.go

```diff
@@ -37,23 +38,46 @@ var (
 	globalCtx context.Context
 )
 
+const (
+	unknownDB   = "Unknown database"
+	createDBDDL = "CREATE DATABASE "
+	mysqlDriver = "mysql"
+)
+
 func closeDB() {
 	if globalDB != nil {
 		globalDB.Close()
 	}
 	globalDB = nil
 }
 
-func openDB() {
+func openDB() error {
 	// TODO: support other drivers
-	dsn := fmt.Sprintf("%s:%s@tcp(%s:%d)/%s", user, password, host, port, dbName)
+	var tmpDB *sql.DB
+	ds := fmt.Sprintf("%s:%s@tcp(%s:%d)/", user, password, host, port)
+	dsn := ds + dbName
 	var err error
-	globalDB, err = sql.Open("mysql", dsn)
+	globalDB, err = sql.Open(mysqlDriver, dsn)
 	if err != nil {
-		panic(err)
+		return err
+	}
+	if err := globalDB.Ping(); err != nil {
+		if strings.Contains(err.Error(), unknownDB) {
+			tmpDB, _ = sql.Open(mysqlDriver, ds)
+			defer tmpDB.Close()
+			if _, err := tmpDB.Exec(createDBDDL + dbName); err != nil {
+				fmt.Printf("failed to create database, err %v\n", err)
+				return err
```

Comment:
return `fmt.Errorf(xxx)` here? And suggest use `panic(err)` below, It's easy to track the error stack.

https://github.com/pingcap/go-tpc/blob/e079372e6af068ca10d03db6b275f93081a901c5/cmd/go-tpc/tpch.go#L16